src/cgutils.cpp

```diff
@@ -988,13 +994,17 @@ static Type *julia_struct_to_llvm(jl_value_t *jt, bool *isboxed)
                 latypes.push_back(lty);
             }
             if (!isTuple) {
-                structdecl->setBody(latypes);
+                if (is_vecelement_type(jt))
+                    // VecElement type is unwrapped in LLVM
+                    jst->struct_decl = latypes[0];
+                else
+                    structdecl->setBody(latypes);
             }
             else {
                 if (isvector && lasttype != T_int1 && !type_is_ghost(lasttype)) {
                     // TODO: currently we get LLVM assertion failures for other vector sizes
-                    bool validVectorSize = (ntypes == 2 || ntypes == 4 || ntypes == 6);
-                    if (0 && lasttype->isSingleValueType() && !lasttype->isVectorTy() && validVectorSize) // currently disabled due to load/store alignment issues
+                    bool validVectorSize = (ntypes == 2 || ntypes == 4 || ntypes == 8 || ntypes == 16);
```

Comment:
I find that many other vector sizes currently work (with LLVM 3.7.1). I'm only having trouble with vector sizes that are of the form `2^n-1` (see my segfault description below).

I think that `VecElement` should always translate to an LLVM vector. If there's a problem with it, then codes that use `VecElement` need to be aware of this anyway, and avoid these vector sizes. I thus think that this check should be simply removed.